src/pkg/reg/adapter/artifacthub/adapter_test.go

```diff
@@ -1,12 +1,35 @@
 package artifacthub
 
 import (
+	"net/http"
+	"os"
+	"testing"
+
 	adp "github.com/goharbor/harbor/src/pkg/reg/adapter"
 	"github.com/goharbor/harbor/src/pkg/reg/model"
 	"github.com/stretchr/testify/assert"
-	"testing"
+	"gopkg.in/h2non/gock.v1"
 )
 
+var ahAdapter *adapter
+
+func init() {
+	var err error
+	ahRegistry := &model.Registry{
+		Type: model.RegistryTypeArtifactHub,
+		URL:  "https://artifacthub.io",
+	}
+	ahAdapter, err = newAdapter(ahRegistry)
+	if err != nil {
+		os.Exit(1)
+	}
+	gock.InterceptClient(ahAdapter.client.httpClient)
+}
```

Comment:
Suggest making these as a function.

Reply:
code has been updated per this review comment.